src/main/scala/edu/arizona/sista/odin/impl/ThompsonVM.scala

```diff
@@ -85,35 +95,35 @@ object ThompsonVM {
     // Instruction is expected to be a Match instruction.
     def stepSingleThread(t: SingleThread): Seq[Thread] = t.inst match {
       case i: MatchToken if t.tok < doc.sentences(sent).size && i.c.matches(t.tok, sent, doc, state) =>
-        mkThreads(t.tok + 1, i.next, t.groups, t.mentions, t.partialGroups)
+        val nextTok = if (t.dir == LeftToRight) t.tok + 1 else t.tok - 1
```

Comment:
After matching a token, the next token is decided based on the thread's direction